src/tests/test_animation/test_animation.cpp

```diff
@@ -383,6 +383,40 @@ private Q_SLOTS:
 		QCOMPARE(a.key_get_frame(2), 40);
 		QCOMPARE(a.key_get_frame(3), -1);
 	}
+
+	void NextKey()
+	{
+		Properties p;
+		p.set("foo", "50=100; 60=60; 100=0");
+		// Cause the string to be interpreted as animated value.
+		p.anim_get_int("foo", 0);
+		Animation a = p.get_animation("foo");
+		QVERIFY(a.is_valid());
+
+		QCOMPARE(a.next_key(0), 50);
+		QCOMPARE(a.next_key(59), 60);
+		QCOMPARE(a.next_key(60), 60);
+		QCOMPARE(a.next_key(61), 100);
+		QCOMPARE(a.next_key(100), 100);
+		QCOMPARE(a.next_key(101), 1);
+	}
+
+	void PreviousKey()
+	{
+		Properties p;
+		p.set("foo", "50=100; 60=60; 100=0");
+		// Cause the string to be interpreted as animated value.
+		p.anim_get_int("foo", 0);
+		Animation a = p.get_animation("foo");
+		QVERIFY(a.is_valid());
+
+		QCOMPARE(a.previous_key(0), 50);
```

Comment:
This seems like a strange return value for this. It is outside of the range, but maybe it should return an error instead of the "next" key?

Reply:
agreed